pom добавление зависимостей

Miogan opened the pull request.

Miogan committed:
pom добавление зависимостей

Miogan committed:
добавление тестов

okryukova commented on src/test/java/ru/practicum/MockitoBurgerTest.java:
```diff
@@ -0,0 +1,344 @@
+    private Bun mockAnotherBun;
+
+    @Mock
+    private Ingredient mockIngredient1;
```
⛔️Нужно исправить. При нейминге не рекомендуется использовать числа (Field2), их еще называют magicNumbers. Очень тяжело поддерживать код с magicNumbers.

Miogan replied:
Исправлено. Параметры переименованы

okryukova commented on src/test/java/ru/practicum/MockitoBurgerTest.java:
```diff
@@ -0,0 +1,344 @@
+        int mayoIndex = receipt.indexOf("майонез");
+
+        // Проверяем, что салат теперь первый, потом помидор, потом майонез
+        assertTrue("Салат должен быть перед помидором", saladIndex < tomatoIndex);
```
⛔️Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде

Miogan replied:
Исправлено

okryukova commented on src/test/java/ru/practicum/MockitoBurgerTest.java:
```diff
@@ -0,0 +1,344 @@
+        String receipt = burger.getReceipt();
+
+        // Проверяем все элементы чека
+        assertTrue("Чек должен содержать название булочки", receipt.contains("белая булочка"));
```
⛔️Нужно исправить. Строку рецепта проверяем целиком

Miogan replied:
Исправлено

okryukova commented on pom.xml:
```diff
@@ -5,12 +5,129 @@
 
     <groupId>org.example</groupId>
-    <artifactId>praktikum</artifactId>
+    <artifactId>diplom</artifactId>
```
⛔️Нужно исправить. Нужно приложить отчет

Miogan replied:
папка site добавлена в гит

Miogan committed:
Исправление замечаний
1. Нужно исправить. При нейминге не рекомендуется использовать числа (Field2), их еще называют magicNumbers. Очень тяжело поддерживать код с magicNumber. Параметры переименованы

Miogan committed:
Исправление замечаний
1. Нужно исправить. Нужно приложить отчет

Miogan committed:
Исправление замечаний
1. Нужно исправить. Для юнит-тестов придерживаемся подхода: один тест, значит одна проверка. Если очень хочется несколько проверок -- тогда используем softAssertions. Поправь, пожалуйста, во всем коде
2. Нужно исправить. Строку рецепта проверяем целиком
